astropy/table/_np_utils.pyx

```diff
@@ -16,7 +16,7 @@ cimport numpy as np
 np.import_array()
 
 DTYPE = int
-ctypedef np.intp_t DTYPE_t
+ctypedef np.int64_t DTYPE_t
```

Comment:
`intp` is correct I think, it's already an int64 on 64-bit cpus, similar to `Py_ssize_t`.
https://numpy.org/devdocs/reference/c-api/dtype.html#c.npy_intp

Reply:
Oh, I think I misread https://numpy.org/doc/stable/release/1.25.0-notes.html#cython-long-t-and-ulong-t-removed and thought its use was discouraged.

Reply:
done